src/routes/api/cryptor-migrations/migration-router.ts

```diff
@@ -0,0 +1,66 @@
+import { Request, Response } from "express";
+import { Installation } from "models/installation";
+
+const MAX_BATCH_SIZE = 10_000;
+const DEFAULT_BATCH_SIZE = 10;
+const RESPONSE_BATCH_SIZE = 1000;
+
+export const CryptorMigrationInstallationPost = async (req: Request, res: Response): Promise<void> => {
+
+	const startTime = Date.now();
+
+	const batchSize = getValidBatchSize(req);
+
+	req.log = req.log.child({ operation: "migrate-installations-shared-secret", batchSize });
+
+	//--------- finding all in current batch to process --------------
+	req.log.info("About to start migrating installation sharedSecret");
+	const installations: Installation[] = await Installation.findAll({
+		limit: batchSize,
+		where: {
+			encryptedSharedSecret: null
+		}
+	});
+	if (installations.length === 0) {
+		req.log.info("No matching batch found, all done, returning now...");
+		res.status(200).send("No matching found, nothing to do");
+		return;
+	}
+	req.log.debug("Found matching installations to migrate, length " + installations.length);
+
+	//--------- loop and save each one of them --------------
+	let count = 0;
+	for (const inst of installations) {
+		//copy from shared secret to encryptedSharedSecret to enable encryption
+		inst.encryptedSharedSecret = inst.sharedSecret;
+		//update db
+		await inst.save();
```

Comment:
no transaction yet, still reading.
but a bonus for no transaction is "resume from where you left last time".

Reply:
@mboudreau let me know if you find out more. 
I just realize the benefit of "resume from where you left last time". Assuming you are running up to 999 out of 1000, we don't want to rollback previous 999 migration as they are independant. Should use transaction as it is designed to be (transaction), not for other like performance purpose.